File: frontend/src/lib/types/admin-blog.ts
// Mirrors backend/app/schemas/admin_blog.py exactly — keep these two in sync.

export type BlogPostStatus = "draft" | "published";

export interface BlogPostCreate {
  title: string;
  category?: string | null;
  tag_key?: string | null;
  author_name?: string | null;
  excerpt?: string | null;
  content_html?: string | null;
  status?: BlogPostStatus;
  featured?: boolean;
  featured_image_id?: string | null;
  tags?: string[];
}

export interface BlogPostUpdate {
  title?: string;
  category?: string | null;
  tag_key?: string | null;
  author_name?: string | null;
  excerpt?: string | null;
  content_html?: string | null;
  featured?: boolean;
  featured_image_id?: string | null;
  tags?: string[];
}

export interface BlogPostAdminSummary {
  id: string;
  title: string;
  category: string | null;
  author_name: string | null;
  status: BlogPostStatus;
  featured: boolean;
  views: number;
  featured_image_url: string | null;
  created_at: string;
  updated_at: string;
}

export interface BlogPostAdminDetail {
  id: string;
  title: string;
  category: string | null;
  tag_key: string | null;
  author_name: string | null;
  excerpt: string | null;
  content_html: string | null;
  status: BlogPostStatus;
  featured: boolean;
  featured_image_id: string | null;
  featured_image_url: string | null;
  tags: string[];
  views: number;
  created_at: string;
  updated_at: string;
}

export const BLOG_CATEGORIES = ["Research", "Data Updates", "Climate", "Policy", "Field Reports"];

File: backend/app/schemas/admin_blog.py
import uuid
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


class BlogPostCreate(BaseModel):
    title: str = Field(min_length=1, max_length=500)
    category: str | None = Field(default=None, max_length=100)
    tag_key: str | None = Field(default=None, max_length=50)
    author_name: str | None = Field(default=None, max_length=255)
    excerpt: str | None = None
    content_html: str | None = None
    status: Literal["draft", "published"] = "draft"
    featured: bool = False
    featured_image_id: uuid.UUID | None = None
    tags: list[str] = Field(default_factory=list)


class BlogPostUpdate(BaseModel):
    title: str | None = Field(default=None, min_length=1, max_length=500)
    category: str | None = Field(default=None, max_length=100)
    tag_key: str | None = Field(default=None, max_length=50)
    author_name: str | None = Field(default=None, max_length=255)
    excerpt: str | None = None
    content_html: str | None = None
    featured: bool | None = None
    featured_image_id: uuid.UUID | None = None
    tags: list[str] | None = None


class BlogPostAdminSummary(BaseModel):
    id: uuid.UUID
    title: str
    category: str | None
    author_name: str | None
    status: str
    featured: bool
    views: int
    featured_image_url: str | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class BlogPostAdminDetail(BaseModel):
    id: uuid.UUID
    title: str
    category: str | None
    tag_key: str | None
    author_name: str | None
    excerpt: str | None
    content_html: str | None
    status: str
    featured: bool
    featured_image_id: uuid.UUID | None
    featured_image_url: str | None
    tags: list[str]
    views: int
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
